pkg/container-utils/containerd/containerd.go

```diff
@@ -15,31 +15,246 @@
+	LabelK8sContainerdKind        = "io.cri-containerd.kind"
+	LabelK8sContainerdKindSandbox = "sandbox"
 )
 
 type ContainerdClient struct {
-	criclient.CRIClient
+	client *containerd.Client
 }
 
 func NewContainerdClient(socketPath string) (runtimeclient.ContainerRuntimeClient, error) {
 	if socketPath == "" {
 		socketPath = runtimeclient.ContainerdDefaultSocketPath
 	}
 
-	criClient, err := criclient.NewCRIClient(runtimeclient.ContainerdName, socketPath, DefaultTimeout)
+	client, err := containerd.New(socketPath,
+		containerd.WithTimeout(DefaultTimeout),
+		containerd.WithDefaultNamespace("k8s.io"),
+	)
 	if err != nil {
 		return nil, err
 	}
 
 	return &ContainerdClient{
-		CRIClient: criClient,
+		client: client,
+	}, nil
+}
+
+func (c *ContainerdClient) Close() error {
+	if c.client != nil {
+		return c.client.Close()
+	}
+	return nil
+}
+
+func (c *ContainerdClient) GetContainers() ([]*runtimeclient.ContainerData, error) {
+	containers, err := c.client.Containers(context.TODO(), "")
+	if err != nil {
+		return nil, fmt.Errorf("listing containers: %w", err)
+	}
+
+	ret := make([]*runtimeclient.ContainerData, 0, len(containers))
+	for _, container := range containers {
+		if isSandboxContainer(container) {
+			log.Debugf("ContainerdClient: container %q is a sandbox container. Temporary skipping it", container.ID())
+			continue
+		}
+
+		task, err := getContainerdTask(container)
+		if err != nil {
+			return nil, err
+		}
+
+		containerData, err := taskAndContainerToContainerData(task, container)
+		if err != nil {
+			return nil, err
+		}
+
+		ret = append(ret, containerData)
+	}
+
+	return ret, nil
+}
+
+func (c *ContainerdClient) GetContainer(containerID string) (*runtimeclient.ContainerData, error) {
+	containerID, err := runtimeclient.ParseContainerID(runtimeclient.ContainerdName, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, _, _, err := c.getContainerDataAndContainerAndTask(containerID)
+	if err != nil {
+		return nil, err
+	}
+	return containerData, nil
+}
+
+func (c *ContainerdClient) GetContainerDetails(containerID string) (*runtimeclient.ContainerDetailsData, error) {
+	containerID, err := runtimeclient.ParseContainerID(runtimeclient.ContainerdName, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containerData, container, task, err := c.getContainerDataAndContainerAndTask(containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := container.Spec(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("getting spec for container %q: %w", containerID, err)
+	}
+
+	mountData := make([]runtimeclient.ContainerMountData, len(spec.Mounts))
+	for i := range spec.Mounts {
+		mount := spec.Mounts[i]
+		mountData[i] = runtimeclient.ContainerMountData{
+			Source:      mount.Source,
+			Destination: mount.Destination,
+		}
+	}
+
+	return &runtimeclient.ContainerDetailsData{
+		ContainerData: *containerData,
+		Pid:           int(task.Pid()),
```

Comment:
`task` might be `nil`, and in case it is, `GetContainerDetails` should return an error `got zero pid` as others `ContainerRuntimeClient` do.